Framework/DataHandling/inc/MantidDataHandling/GroupDetectors2.h

```diff
@@ -205,7 +205,8 @@ class DLLExport GroupDetectors2 : public API::Algorithm {
   /// Copy the and combine the histograms that the user requested from the input
   /// into the output workspace
   size_t formGroups(API::MatrixWorkspace_const_sptr inputWS,
-                    API::MatrixWorkspace_sptr outputWS, const double prog4Copy);
+                    API::MatrixWorkspace_sptr outputWS, const double prog4Copy,
+                    const bool keepAll, const std::set<int64_t> &unGroupedSet);
   /// Copy the and combine the event lists that the user requested from the
   /// input into the output workspace
   size_t formGroupsEvent(DataObjects::EventWorkspace_const_sptr inputWS,
```

Comment:
Should the same changes also be done for `formGroupsEvent`?

Edit: Actually that might be tricky, since `EventList::operator+=` handles merging of detector ID groups. So maybe it is ok to not handle that case for now. Should we at least throw an exception?